### PR TITLE
[CVP-1718]-add ansible/yaml lint

### PR DESCRIPTION
**Pull request by jsztuka**
Closes CVP-1718
Few information regarding those changes.
Ansible and yaml lint config files used from CVP downstream.
Yamllint config remains the same, however ansible-linting ocnfig needes some updates since there is used the newer version (5.0.5).
Most significant change in the config is this line: 
`warn_list:
            - 'internal-error'`
It is taking care of this part : https://github.com/redhat-operator-ecosystem/operator-test-playbooks/blob/master/deploy-olm-operator.yml#L7
on which it would fail due to nonexisting file reference. Rest of the added skip_list rules are not that significant.
I also tried implementing Ansible-lint github action, which is probably out of date, I was unable to get this thing running, even with the very basic test, it was unable to find certain files to execute correctly.
As you can see the yamllint output is not verbose enought, so we can't see exactly where the issue is, github needs to be configured to have more verbose output(debug).
I am open to any suggestions/proposals regarding the code.

### CODE REVIEW

**Comment by jsztuka** [comment]
Issues with : CVP/cvp-redhat-operator-bundle-image-validation-test
and CVP/cvp-redhat-operator-metadata-validation-test
are not caused by the changes.

**Comment by samvarankashyap** [comment]
@jsztuka There is a syntax error on  deploy-olm-operator.yml 
from the log. Could you fix it, please.

**Comment by lslebodn** [comment]
@jsztuka "CVP/cvp-redhat-operator-metadata-validation-test" passed in the
other PR https://github.com/redhat-operator-ecosystem/operator-test-playbooks/pull/132

**Comment by jsztuka** [comment]
From jenkins logs: 
>>Cancelling nested steps due to timeout<<

**Comment by jsztuka** [comment]
> @jsztuka "CVP/cvp-redhat-operator-metadata-validation-test" passed in the
> other PR #132

Experiencing the same issue also:
From #132 ci pipeline run
java.lang.Exception: Error running Ansible playbook 'product/playbooks/trigger-and-wait-for-product-build.yml': org.jenkinsci.plugins.workflow.steps.FlowInterruptedException

**Comment by lslebodn** [comment]
@jsztuka  I can see following error in log but test is marked as succesful
```
Error: internal-error Unexpected error code 1 from execution of: ansible-playbook --syntax-check deploy-olm-operator.yml
You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
Finished with 0 failure(s), 1 warning(s) on 15 files.
```

Why do we ignore errors?

**Comment by jsztuka** [comment]
Figure out issue with not having files included in output of the yaml_lint.

**Comment by jsztuka** [comment]
> @jsztuka I can see following error in log but test is marked as succesful
> 
> ```
> Error: internal-error Unexpected error code 1 from execution of: ansible-playbook --syntax-check deploy-olm-operator.yml
> You can skip specific rules or tags by adding them to your configuration file:
> # .ansible-lint
> warn_list:  # or 'skip_list' to silence them completely
> Finished with 0 failure(s), 1 warning(s) on 15 files.
> ```
> 
> Why do we ignore errors?

We need to take care of this approach, so we don't require files from downstream.

File: .github/workflows/yaml_and_ansible_linting.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: "Ansible-lint"
+      # Install ansible-lint together with yamllint and run the linting on the whole repo
+        run: |
+          sudo pip install "ansible-lint[yamllint]"
```
Comment (lslebodn) [inline .github/workflows/yaml_and_ansible_linting.yml:27]: Can you pin version of packages from pypi?

Comment (jsztuka) [reply .github/workflows/yaml_and_ansible_linting.yml:27]: Sure thing.

Comment (jsztuka) [reply .github/workflows/yaml_and_ansible_linting.yml]: Is that format sufficient, our you want to set each one individually?

File: deploy-olm-operator.yml [hunk 2]
```diff
@@ -4,10 +4,9 @@
   become: false
   gather_facts: false
 
-  vars_files:
-    - "{{ playbook_dir }}/../../../config/ansible/vault.yml"
 
   vars:
+    vault_file_path: "{{ vault_file_path | default('{{ playbook_dir }}/../../../config/ansible/vault.yml')}}"
```
Comment (samvarankashyap) [inline deploy-olm-operator.yml:9]: @jsztuka Wrong syntax will cause errors in downstream it should be as follows: 
```
vault_file_path: "{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}"
```

**Review by samvarankashyap — DISMISSED** [review]

File: deploy-olm-operator.yml (hunk 2 above)
Comment (lslebodn) [reply deploy-olm-operator.yml:9]: @jsztuka  bump

Comment (jsztuka) [reply deploy-olm-operator.yml:9]: Thanks for notice!

**Review by dirgim — APPROVED** [review]
LGTM

File: deploy-olm-operator.yml [hunk 3]
```diff
@@ -4,17 +4,16 @@
   become: false
   gather_facts: false
 
-  vars_files:
-    - "{{ playbook_dir }}/../../../config/ansible/vault.yml"
 
   vars:
+    vault_file_path: "{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}"
```
Comment (lslebodn) [inline deploy-olm-operator.yml:9]: @jsztuka it is still failing
```
{'allow_duplicates': <ansible.playbook.attribute.FieldAttribute object at 0x7fd8d345ba90>, 'dependencies': <ansible.playbook.attribute.FieldAttribute object at 0x7fd8d345bac0>}, '_dependencies': [], '_parents': []}}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path.quay_token }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | default(playbook_dir +'/../../../config/ansible/vault.yml') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ vault_file_path | 
```

File: deploy-olm-operator.yml [hunk 4]
```diff
@@ -4,17 +4,16 @@
...
     run_upstream: false
     openshift_namespace: "test-operator"
     operator_work_dir: "/home/jenkins/agent/test-operator"
     scorecard_cr_dir: "/home/jenkins/agent/scorecard-cr-files"
     work_dir: "{{ lookup('env', 'WORKSPACE') }}"
     olm_operator_files_path: "{{ work_dir }}/olm-operator-files"
-    quay_token: "{{ cvp_vault.quay_token }}"
+    quay_token: "{{ vault_file_path.quay_token }}"
```
Comment (lslebodn) [inline deploy-olm-operator.yml:16]: @jsztuka if we need to load wjole vaout file jsut for `quay_token` than it seems to me like overkill.
We can pass quay token with `ansible-playbook --extra-vars`

**Review by lslebodn — COMMENTED** [review]
still issue with vault

File: deploy-olm-operator.yml (hunk 4 above)
Comment (jsztuka) [reply deploy-olm-operator.yml]: --extra-vars is more appealing for me, we don't use any other secret besides this one in the playbook.

**Comment by samvarankashyap** [comment]
@jsztuka All looks fine for me but 
i see the following error at many places that 
/ build

roles/collect_modified_images/tasks/main.yml#L41

risky-file-permissions File permissions unset or incorrect

**Review by samvarankashyap — COMMENTED** [review]
Check errors here 
https://github.com/redhat-operator-ecosystem/operator-test-playbooks/pull/134/files

**Comment by jsztuka** [comment]
Do we really want to add file permissions at all accurences that are marked by Github?

**Comment by jsztuka** [comment]
> Check errors here
> https://github.com/redhat-operator-ecosystem/operator-test-playbooks/pull/134/files

Fixed.

**Comment by jsztuka** [comment]
MR https://gitlab.sat.engineering.redhat.com/cvp/pipeline/-/merge_requests/1761 Needs to merged prior this change.
Blocked on PSI outage.

**Review by dirgim — APPROVED** [review]
LGTM

File: .github/workflows/yaml_and_ansible_linting.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: "Ansible-lint"
+      # Install ansible-lint(5.0.6) together with yamllint(1.26.0) and run the linting on the whole repo
+        run: |
+          sudo python -m pip install ansible
```
Comment (mvalarh) [inline .github/workflows/yaml_and_ansible_linting.yml:27]: I think `sudo` is not needed here. You are running as `root`.

Comment (jsztuka) [reply .github/workflows/yaml_and_ansible_linting.yml:27]: Yep, just tested it locally, there is no need for sudo to be here.

**Review by mvalarh — APPROVED** [review]

**Review by samvarankashyap — COMMENTED** [review]
@jsztuka I still see risky file permissions to some of the files. 
could you change them ?

**Comment by jsztuka** [comment]
> @jsztuka I still see risky file permissions to some of the files.
> could you change them ?

Havent seen them before, fixing them now.

**Comment by jsztuka** [comment]
Github doesn't show all the risky file permissions at once, this one will take a while.

**Review by lslebodn — CHANGES_REQUESTED** [review]
```
WARNING  Listing 1 violation(s) that are fatal
AnsibleUndefined
AnsibleUndefined
risky-file-permissions: File permissions unset or incorrect
roles/extract_operator_bundle/tasks/main.yml:7 Task/Handler: Ensure that the operator testing directory exists and is empty

Error: risky-file-permissions File permissions unset or incorrect
You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental
Finished with 0 failure(s), 1 warning(s) on 74 files.
```

@jsztuka Could you explain why the issue is reported but all checks are green ?

**Comment by jsztuka** [comment]
This is a run of the pending job:
https://jenkins-cvp-stage-ci-operatortestplaybooks.cloud.paas.psi.redhat.com/job/cvp-redhat-operator-bundle-image-validation-test/24/

**Comment by jsztuka** [comment]
> ```
> WARNING  Listing 1 violation(s) that are fatal
> AnsibleUndefined
> AnsibleUndefined
> risky-file-permissions: File permissions unset or incorrect
> roles/extract_operator_bundle/tasks/main.yml:7 Task/Handler: Ensure that the operator testing directory exists and is empty
> 
> Error: risky-file-permissions File permissions unset or incorrect
> You can skip specific rules or tags by adding them to your configuration file:
> # .ansible-lint
> warn_list:  # or 'skip_list' to silence them completely
>   - experimental  # all rules tagged as experimental
> Finished with 0 failure(s), 1 warning(s) on 74 files.
> ```
> 
> @jsztuka Could you explain why the issue is reported but all checks are green ?

Risky file permissions are implicitly only set as a warning, hence the error appeared. They are not causing ansible-lint to fail.
However the error should be gone now, could you confirm?

**Comment by jsztuka** [comment]
Bundle-image-validation ended up with success:
https://jenkins-cvp-ci.cloud.paas.psi.redhat.com/job/test-operator-test-playbooks/457/console

**Comment by lslebodn** [comment]
> Risky file permissions are implicitly only set as a warning, hence the error appeared. They are not causing ansible-lint to fail.
> However the error should be gone now, can you confirm?

But looks like github reported warnings as comments to the code.
and than SK requested changes. What we can do to improve UX here?

**Comment by jsztuka** [comment]
> > Risky file permissions are implicitly only set as a warning, hence the error appeared. They are not causing ansible-lint to fail.
> > However the error should be gone now, can you confirm?
> 
> But looks like github reported warnings as comments to the code.
> and than SK requested changes. What we can do to improve UX here?

We need to make sure that ansible-lint register those issues as errors and not as warnings.

**Comment by jsztuka** [comment]
Ansible-lint is now handling warnings as errors:
https://github.com/redhat-operator-ecosystem/operator-test-playbooks/pull/134/checks?check_run_id=2342658925

**Review by lslebodn — APPROVED** [review]
lgtm

**Pull request merged**